File: lclang/src/main/java/lclang/exceptions/LCLangException.java
package lclang.exceptions;

import lclang.Caller;

public class LCLangException extends Exception {
    public LCLangException(String name, String message, Caller caller) {
        super(name+": "+message+"\n\t"+String.join("\n\t", caller.toStacktrace()));
    }
}

File: lclang/src/main/kotlin/lclang/exceptions/LCLangException.kt
package lclang.exceptions

import lclang.Caller

open class LCLangException(exceptionName: String, message: String, caller: Caller):
    Exception("$exceptionName: $message\n"+caller.toList().reversed().joinToString("\n") {
        "\tat ${it.root.path}:${it.line} on line ${it.line}, column ${it.column}"
    })